src/test/java/com/hazelcast/commandline/member/MemberCommandLineTest.java

```diff
@@ -16,190 +16,167 @@
+        memberCommandLine = new MemberCommandLine(out, mock(PrintStream.class), hazelcastProcessStore, processExecutor, false);
     }
 
-    private void killAllRunningHazelcastInstances() {
-        try {
-            for (HazelcastProcess hazelcastProcess : memberCommandLine.getProcessStore().findAll().values()) {
-                runCommand("kill -9 " + hazelcastProcess.getPid());
-            }
-        } catch (Exception e) {
-            //ignored, test instances file might not exist.
-        }
-    }
-
-    @Test(timeout = 10000)
+    @Test
     public void test_start()
             throws IOException, InterruptedException {
-        memberCommandLine.start(null, DEFAULT_CLUSTER_NAME, DEFAULT_PORT, null, false, null, null);
-        Stream<String> processOutput = getProcessOutput(memberCommandLine.getProcessInputStream());
-        assertTrue(processOutput.anyMatch(out -> out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+        //when
+        memberCommandLine.start(null, null, null, null, false, null, null);
+        //then
+        verify(hazelcastProcessStore, times(1)).create();
+        verify(processExecutor, times(1)).buildAndStart(anyList(), eq(false));
+        verify(hazelcastProcessStore, times(1)).save(any());
     }
 
-    @Test(timeout = 10000)
+    @Test
     public void test_start_withConfigFile()
-            throws IOException, InterruptedException {
-        String groupName = "member-command-line-test";
-        startMemberWithConfigFile();
-        Stream<String> processOutput = getProcessOutput(memberCommandLine.getProcessInputStream());
-        assertTrue(processOutput
-                .anyMatch(out -> out.contains(groupName) && out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+            throws Exception {
+        // given
+        String configFile = "path/to/test-hazelcast.xml";
+        // when
+        memberCommandLine.start(configFile, null, null, null, false, null, null);
+        // then
+        verify(processExecutor)
+                .buildAndStart((List<String>) argThat(Matchers.hasItem("-Dhazelcast.config=" + configFile)), eq(false));
     }
 
-    @Test(timeout = 10000)
+    @Test
     public void test_start_withClusterName()
-            throws IOException, InterruptedException {
-        String groupName = "member-command-line-test";
-        memberCommandLine.start(null, groupName, DEFAULT_PORT, null, false, null, null);
-        Stream<String> processOutput = getProcessOutput(memberCommandLine.getProcessInputStream());
-        assertTrue(processOutput
-                .anyMatch(out -> out.contains(groupName) && out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+            throws Exception {
+        // given
+        String clusterName = "member-command-line-test";
+        // when
+        memberCommandLine.start(null, clusterName, null, null, false, null, null);
```

Comment:
There's a number of these constructor invocations in tests and they don't really look easy to read.
Would you consider adding a builder object (e.g., `memberCommandLine.builder().clusterName(clusterName).start()`)?
Or at least a small number of static factory methods in a test utility.
(I believe the multi-arg constructor cannot go away as it's needed by picocli...)

Reply:
The issue is about picocli requirements here, as you stated. If I build a builder object, we also need to write unit tests for it, which will end in a similar issue, right? Because structure of the builder class can be changed without any test failures, if we don't add unit tests for it. Wdyt?

Reply:
That's a very good question :-) we should look into best practices about testing builders! Back to the original comment, maybe we could just have static shortcut methods for better readability, such as 
`startWithClusterName(memberCommandLine, clusterName)` or `startWithDefaultArguments(memberCommandLine)` etc...

Reply:
I tried adding new shortcut methods, but it didn't help that much in readability since the names of the test method already mention the purpose. I will keep it as it for now.

Reply:
Fair enough :-)